### DISCUSSION TITLE
how exactly does snap_entropy works?

### DISCUSSION THREAD

**Discussion by Arch0125** [MetaMask/snaps#1444] [Q&A] [answered]
I've recently used snap_entropy to generate pvt keys and use them as signers, i would like to know what exactly is used for generating the key pair and is it anyway derived from the seedphrase

**Comment by ziad-saab** [comment, marked as answer]
This explains how `snap_getEntropy` works: https://docs.metamask.io/snaps/reference/rpc-api/#snap_getentropy

**Comment by Montoya** [comment]
If you want a deeper look into how it is done, you can look at our key-tree package: https://github.com/MetaMask/key-tree
